### ISSUE TITLE
Phantasy Star Online Episode I & II Plus JP: Can't import devolution memcard.bin game saves

### ISSUE THREAD

**Issue by edwinbradford** [FIX94/Nintendont#849]
I'm unable to import my PSO Plus (Japan) `memcard.bin` game saves from Devolution.

Steps to reproduce:

1. Rename the Devolution `memcard.bin` file to `GPOJ.raw`.
2. Create a `saves` folder in the root of the USB and place `GPOJ.raw` in this folder.
3. Set `Memcard Emulation: On`, `Memcard Blocks: 251` and `Memcard Multi: Off`.
4. Select the PSO game and launch it.

Expected result:
The game recognises the saved data and loads it.

Result:
A message in Japanese (I can read Japanese) says the data is corrupt and do I want to format it. Selecting No will ask you the same question in a loop from which you cannot exit. Selecting Yes takes you to the Calendar menu screen at which point I understand you should select No. On selecting No the game runs but without loading any saved data.

- Nintendont version 6.490
- Nintendont app, save data and disc are all on the USB
- The original memcard.bin is attached

[memcard.zip](https://github.com/FIX94/Nintendont/files/5754739/memcard.zip)

**Comment by carnage702** [comment]
first update nintendont to 491

second you cant just rename stuff and make it work. you need to extract the save file somehow, im not usre how devolution saves to be honest but if dolphin can open it you can do it.

1- delete any pso save file from nintendont
2- start pso and create a new save file then shutdown console
3-get the jap pso save raw file and copy it to your pc
4- open dolphin memory card manager tap and open both nintendont save file and devolution(if its supported that is)
5-delete the nintendont save file  from the nintendont raw file and copy the devolution save file inside it and save.
6-copy the save file back to the usb/sd saves folder and profit?

**Comment by hurda** [comment]
Renaming it to memcard.raw allows Dolphin to open it as a raw memcard-dump, showing five saves (three for PSO, one other game, one "Sega Network Information"), so Nintendont probably should be able to open it, too.

**Comment by edwinbradford** [comment]
I'm still working on it... unfortunately at the PSO "Do you want to format this memory card screen?" I pressed Home on the Wii controller to exit the game, shut down the Wii then removed the USB stick and my PC no longer recognises it. My other PC has no problem so I'm in the middle of reformatting the USB stick and restoring it assuming it's still okay.

Before it died I deleted `MemoryCardA.JAP.raw` that I found in the Dolphin GC directory and replaced it with the `memcard.bin` file from Devolution which I renamed back to `MemoryCardA.JAP.raw`. I launched PSO and Dolphin had no problem recognising the game saves.

It appears therefore that renaming `memcard.bin` to `GPOJ.raw` should be all that's required according to the Nintendont documentation. @hurda the 59 block 251 block conflict is a great suggestion. I'll look into it as soon as I get my USB or another USB stick up and running.

**Comment by edwinbradford** [comment]
I got it working.

The corupted USB stick was possibly unrelated... my USB 2.0 to 3.0 PCIE card in my PC stopped recognising USB sticks and required a complete PC shut down and reboot.

The steps I took to export Devolution PSO game saves to Nintendont were:

1. Create a new empty [59] block memory card in Dolphin Memory Card Manager and name it `GPOJ.raw`.
2. Rename the Devolution `memcard.bin` save file to `mycard.raw`.
3. In Memory Card Manager open `mycard.raw` select all PSO files and copy them to `GPOJ.raw`.
4. Copy `GPOJ.raw` to the root of your USB stick.
5. In the Nintendont settings set `Memcard Blocks: [59]`.
6. Launch the game.

As long as the [block size] set in Dolphin Memory Card Manager and Nintendont Settings are the same any block size should work in theory. The name of the raw file for for PSO Plus Japan edition in Nintendont must be "GPOJ" see the documentation. Other versions of PSO (other games) require a different four letter code which can be found from [redump.org](http://redump.org/). 

Thanks @hurda and @carnage702, I wouldn't have figured it out without your help.
